### ISSUE TITLE
Remove need for manual CI trigger on pull requests

### ISSUE THREAD

**Issue by ndmitchell** [digital-asset/ghcide#23]
The /azp run thing is pretty annoying for contributing. It slows down the time to feedback, introduces a two-touch minimum from DA employees etc. Any reason not to go with standard Appveyor/Travis more like most open source projects? Or Azure nodes not hosted internally?

**Comment by cocreature** [comment]
We are already not using internal nodes and we’re working on trying to get rid of the `azp run` restriction.

**Comment by cocreature** [comment]
For now, I’d like to stick with a non-nix based setup since most engineers internally don’t have much nix experience (myself not included) but thank you very much for the offer!

**Comment by domenkozar** [comment]
Could run two CIs in parallel, my goal is to not maintain the Nix expressions separately since that brings cognitive overhead for users as well as manual work :)

**Comment by cocreature** [comment]
Adding a separate CI job seems reasonable but I have admittedly zero experience with hercules-ci or any idea what that would entail.

**Comment by domenkozar** [comment]
Essentially you need to run a daemon on your own hardware. See https://docs.hercules-ci.com/hercules-ci/getting-started/

I'm happy to run it on our infrastructure if that's too much hassle for you.

**Comment by cocreature** [comment]
Running it on our own infrastructure is not an option, that was part of the reason why our CI currently does not run automatically for external contributors (I know Nix has sandboxing but it’s not my decision and not a fight I’m interested in fighting). If we can run it on your infrastructure, that would be an option. I’m slightly worried about adding a CI job based on Nix given the aforementioned unfamiliarity with Nix but I guess the worst that could happen is that we realize that we don’t have the developer resources to maintain it at the moment and we turn it off again, so I’d be happy to try it.

@garyverhaegen-da should probably also weigh in on that.

**Comment by domenkozar** [comment]
I'd recommend running it on a machine not connected to the rest of your code, then using something like https://app.bors.tech/ to test the PRs.

Let me know if you'd like to give it a try, or I can set it up as well.

**Comment by garyverhaegen-da** [comment]
Can we please split the discussion? There are two very different issues here:
- To @ndmitchell 's point, our current restriction that CI requires manual intervention from a DA employee is our default security policy as a company; there are ongoing internal discussions as to whether we can relax that specifically for this repo, but I am not in a position to promise anything regarding their outcome at the moment. To clarify: this is a deliberate decision to impose this restriction on top of whatever CI infrastructure we use, and is therefore not linked to the choice of Azure vs. Appveyor or Travis or any other.
- @domenkozar seems to have a completely different agenda here; it is not clear to me what hercules is (I'm not terribly familiar with the Haskell or nix ecosystems), nor what we would gain from supporting it. Nor why we'd need to; there seems to be a perfectly viable system in plae already over at https://github.com/hercules-ci/ghcide-nix. @domenkozar If you want to pursue this further please open a separate issue explaining why you think we'd want it and what we'd need to do, assuming that changing infrastructure provider is not an option at this point (i.e. it needs to run on Azure).

**Comment by domenkozar** [comment]
@garyverhaegen-da the two agendas meet if the CI switches to infrastructure that you don't use while providing binaries for people to install. Assuming that you're OK with that.

I wanted to see if we can kill two birds with the same stone, although I see your point, so I've opened an issue at https://github.com/digital-asset/ghcide/issues/56

**Comment by garyverhaegen-da** [comment]
@domenkozar Thanks! I'll keep that conversation going over there.

@ndmitchell We have received the authorization to remove the restriction on external PRs for this repo. Does that address your issue?

**Comment by ndmitchell** [comment]
@garyverhaegen-da indeed it does!
